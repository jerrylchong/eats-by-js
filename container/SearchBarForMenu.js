import React from 'react';
import { StyleSheet, TextInput, View, Image, TouchableOpacity, Dimensions } from 'react-native';

class SearchBarForMenu extends React.Component {
    render() {
        const {searchTerm, clearSearch, navigation} = this.props
        return (
            <View style = {styles.container}>
                <View style = {styles.searchBar}>
                    <TextInput
                        style = {styles.input}
                        placeholder = "Search"
                        placeholderTextColor = '#404040'
                        onChangeText = {this.handleSearchTerm}
                        value = {searchTerm} />
                    <TouchableOpacity
                        style = {{height: 40, width: 40, justifyContent: 'center', alignItems: 'center'}}
                        onPress = {clearSearch}>
                        <Image style = {styles.image} source ={require('../assets/X.png')} />
                    </TouchableOpacity>
                </View>
            </View>
        )
    }
}

export default SearchBarForMenu

const windowWidth = Dimensions.get('window').width;

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        width: '90%',
        marginBottom: '3%'
    },
    searchBar: {
        flexDirection: 'row',
        width: windowWidth * 0.9,
        height: windowWidth * 0.1,
        justifyContent: 'space-between',
        alignItems: 'center',
        backgroundColor: '#ececec',
        borderRadius: 25,
    },
    back: {
        height: 50,
        width: 30,
        justifyContent: 'center'
    },
    backImage: {
        height: 13,
        width: 13
    },
    input: {
        color: '#404040',
        fontSize: 16,
        marginLeft: '5%',
        fontFamily: 'Ubuntu-Light',
        width: '80%'
    },
    image: {
        marginRight: '5%',
        height: 13,
        width: 13
    },
    bigImage: {
        height: 20,
        width: 24
    }
})