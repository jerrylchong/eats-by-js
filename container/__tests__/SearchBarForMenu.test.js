import React from 'react';
import { TouchableOpacity } from "react-native";
import renderer from 'react-test-renderer';
import SearchBarForMenu from "../SearchBarForMenu";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

test('renders correctly', () => {
    const tree = renderer.create(
        <SearchBarForMenu
            searchTerm={''}
        />).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });
test('test clearSearch functionality', () => {
    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('onPress invoked');
    const wrapper = shallow(<SearchBarForMenu searchTerm={''} clearSearch={onPressEvent}/>);
    wrapper.find(TouchableOpacity).props().onPress();
    expect(onPressEvent.mock.calls.length).toBe(1);
})