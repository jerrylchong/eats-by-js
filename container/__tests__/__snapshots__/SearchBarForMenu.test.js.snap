// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": "3%",
        "width": "90%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ececec",
        "borderRadius": 25,
        "flexDirection": "row",
        "height": 75,
        "justifyContent": "space-between",
        "width": 675,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="Search"
      placeholderTextColor="#404040"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#404040",
          "fontFamily": "Ubuntu-Light",
          "fontSize": 16,
          "fontWeight": "normal",
          "marginLeft": "5%",
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 13,
            "marginRight": "5%",
            "width": 13,
          }
        }
      />
    </View>
  </View>
</View>
`;
